File: ething/webui/src/meta/MihomeSensorHT.js
(function (root, factory) {
    if (typeof define === 'function' && define.amd) {
        // AMD. Register as an anonymous module.
        define(['jquery', 'ething', 'form'], factory);
    }
}(this, function ($, EThing) {
	
	return {
		
		'bases': ['MihomeDevice'],
		
		'name': 'MiHome Humidity Temperature and Pressure sensor',
		
		'creatable' : false
		
	}
}))

File: ething/mihome/MihomeSensorHT.py
# coding: utf-8



from .MihomeDevice import MihomeDevice
from ething.interfaces import Thermometer

class MihomeSensorHT(MihomeDevice, Thermometer):
    """
    Mihome temperature/humidity/pressure Sensor Device class.
    """
    
    def processAttr(self, name, value):
        
        if name == 'temperature':
            value = int(value)/100.0
        
        if name == 'humidity':
            value = int(value)/100.0
        
        if name == 'pressure': # hPa
            value = int(value)
        
        self.store(name, value)
    
    
    
    




 